awscli/customizations/ec2/paginate.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+def register_ec2_page_size_injector(event_emitter):
+    EC2PageSizeInjector().register(event_emitter)
+
+
+class EC2PageSizeInjector(object):
+
+    # Operations to auto-paginate and their specific whitelists.
+    # Format:
+    #    Key:   Operation
+    #    Value: List of parameters to add to whitelist for that operation.
+    TARGET_OPERATIONS = {
+        "describe-volumes": [],
+        "describe-snapshots": ['OwnerIds', 'RestorableByUserIds']
+    }
+
+    # Parameters which should be whitelisted for every operation.
+    GLOBAL_WHITELIST = ['NextToken', 'DryRun', 'PaginationConfig']
```

Comment:
There's no problem if it is excluded, but that list is the parameters where there will be no issue putting the paging param in. I'm 0 on keeping it, so I'll remove it if you'd like.